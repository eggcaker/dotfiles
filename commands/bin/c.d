#!/usr/bin/env python3

import pendulum as pt
import sys

now = pt.now()

if (len(sys.argv) == 1):
    print(int(now.timestamp()))
else:
    if (sys.argv[1].isdigit()):
        time_stamp = int(sys.argv[1])
        time_stamp = time_stamp if time_stamp < 6489546000 else time_stamp / 1000
        print(now.fromtimestamp(time_stamp))
    else:
        print(int(now.timestamp()))
