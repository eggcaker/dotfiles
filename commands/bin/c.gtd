#!/usr/bin/env zsh


action="$1";

if [[ $action == "" ]] ; then
    exit 0;
fi

clock_in="(pelm-org/clock-in-task-by-id \"gtd_${action}\")"
emacsclient -e $clock_in
echo "clocked on gtd $action";
