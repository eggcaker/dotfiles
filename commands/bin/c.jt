#!/bin/bash
mkdir $1; 
cd $1; 
mkdir -p src/main/java/com/yep8;
touch build.gradle
echo "apply plugin: 'java'" >> build.gradle
echo "apply plugin: 'application'" >> build.gradle
echo 'mainClassName = "com.yep8.SandBox"' >> build.gradle

echo "repositories {
  mavenCentral()
}

dependencies {
  compile 'joda-time:joda-time:2.8.1'
}

" >> build.gradle


cd src/main/java/com/yep8;

touch SandBox.java
echo "package com.yep8;" >> SandBox.java
echo "import java.util.*;" >> SandBox.java;
echo "import org.joda.time.*;" >> SandBox.java;


echo "

public class SandBox {
  public static void main(String[] args) {
    TestClass tc = new TestClass();
    System.out.println(tc.sayHello());
  }
}" >> SandBox.java

touch TestClass.java
echo "package com.yep8;" >> TestClass.java

echo 'public class TestClass{
  public String sayHello() {
    return "Hello,World";
  }
}' >> TestClass.java


cd -

git setup

cd src/main/java/com/yep8;

emacsclient -n SandBox.java
