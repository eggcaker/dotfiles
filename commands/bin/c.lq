#!/bin/zsh

website_url="http://www.baidu.com/s?ie=utf-8&wd=双色球开奖结果"
wget -O tmp.html $website_url  > /dev/null 2>&1


start_line="`cat tmp.html|grep -n 'class="op_caipiao_table'|awk -F: '{print $1}'`"
total_line="`cat tmp.html|wc -l`"
cut_line="`expr $total_line - $start_line + 2`"

tail -n $cut_line tmp.html  |head -n 1  > tmp2.html

w3m -dump tmp2.html  > tmp.html
result="`cat tmp.html |head -n 3 |tail -n 1 `"
rm tmp.html
rm tmp2.html
echo $result



