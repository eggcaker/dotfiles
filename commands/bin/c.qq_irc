#!/usr/bin/env zsh


/usr/bin/perl -MMojo::Webqq -e 'Mojo::Webqq->new(qq=>$ARGV[0])->login->load(["ShowMsg","IRCShell","StockInfo"])->run()' 3832907
