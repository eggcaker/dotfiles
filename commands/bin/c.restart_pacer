c.stop_pacer
c.start_pacer
