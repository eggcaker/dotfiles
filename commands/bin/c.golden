#!/usr/bin/env bash

url="http://www.bullion-rates.com/gold/CNY/spot-price.htm";

if [ "$1" != "" ]; then
    url="http://www.bullion-rates.com/silver/USD/spot-price.htm"
fi

curl -s  $url > cilver.html

w3m -dump -cols 500 cilver.html > tmp.txt
cat tmp.txt|head -n 13|tail -n 8 |sed -e 's/\[br_icon\] Bullion Prices//g' | \
    sed -e 's/\[bullet\]Silver Prices//g' | \
    sed -e 's/\[bullet\]Gold Prices//g' | \
    sed -e 's/\[bullet\]Platinum Prices//g' | \
    sed -e 's/\[bullet\]Palladium Prices//g' | \
    awk 'NF'

rm cilver.html
rm -rf tmp.txt
